packages/plugin-types/API.md

```diff
@@ -118,6 +118,11 @@ export type ConstructFactoryGetInstanceProps = {
 // @public
 export type DeploymentType = 'branch' | 'sandbox';
 
+// @public
+export type Expand<T> = T extends infer O ? {

```

Comment:
I'm not sure this belongs in `plugin-types`. Probably best to just keep this in `backend-auth` until we know we need it in several places

Reply:
didn't seem specific to auth so I was agreeing with this comment https://github.com/aws-amplify/amplify-backend/pull/606#discussion_r1387323873 from the old PR

Reply:
I suppose I can disagree and commit on this one, but we need to be careful that `plugin-types` doesn't become a dumping ground for all sorts of utility types. Perhaps we need a `utility-types` package?

Reply:
Creating separate set of types for factory and construct removes this problem.
Which btw makes me thing that if we keep current amount of type algebra for now then maybe it's better to move this type back to auth so it's easier to remove it when preview customers say that they can't understand our auth types.

Reply:
I'll move this back to auth